homeassistant/components/scrape/sensor.py

```diff
@@ -75,28 +76,54 @@ async def async_setup_platform(
     discovery_info: DiscoveryInfoType | None = None,
 ) -> None:
     """Set up the Web scrape sensor."""
-    name: str = config[CONF_NAME]
-    resource: str = config[CONF_RESOURCE]
+    _LOGGER.warning(
+        # Config flow added in Home Assistant Core 2022.5, remove import flow in 2022.7
+        "Loading Scrape via platform setup has been deprecated in Home Assistant 2022.5"
+        "Your configuration has been automatically imported and you can"
+        "remove it from your configuration.yaml"
+    )
+
+    if config.get(CONF_VALUE_TEMPLATE):
+        template: Template = config[CONF_VALUE_TEMPLATE]
+        config[CONF_VALUE_TEMPLATE] = template.template
+
+    hass.async_create_task(
+        hass.config_entries.flow.async_init(
+            DOMAIN,
+            context={"source": SOURCE_IMPORT},
+            data=config,
+        )
+    )
```

Comment:
Is there any risk of duplicate imports here?

Reply:
It could be but it's handled in the config_flow and would abort if such.